### DISCUSSION TITLE
Is there a way to queue room cleaning commands?

### DISCUSSION THREAD

**Discussion by tamueller** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#642] [Q&A]
I would love to be able to issue several commands to my Roborock S8 and have them queued somehow. Some people wrote queues in NodeRed, but it would be nice, to use it from this UI.
The reason is, that I want the bot to first vacuum a room, and then mop it - wich is not possible with the out of the box config of this card.
Is there an "easy way" to do it? Or even a complicated one, where I send all the commands this card issues to some other place (e.g. node red via MQTT or something) instead of to the robot itself and due the queueing there myself)

**Comment by PiotrMachowski** [comment]
The card can execute any service call, but the queue itself would have to be made using scripts and automations or node-red. I think the easiest way to achieve it might be using a script with mode `queued`

**Reply by tamueller** [reply]
As someone who is familiar with homeassistant automations and nodered, but has never touched Lovelace cards before: Where would be a starting point to make the card send the bot-commands to a queue instead of the roborock integration? Can I do it in the HA UI where I edit the card (if so, where?) or do I need to edit the sourcecode of the card?

**Reply by PiotrMachowski** [reply]
Service call made by the card can be altered using `map_modes` -> ... -> `service_call_schema`.

You can prepare a parametric service call that will consume room ID and can be called using Dev Tools and then ask me for help in integrating it with the card ;)

**Reply by tamueller** [reply]
Thanks for the help.
I just tired to use the "generate static config" button, to see how the code looks like, and I think I can alter it to go to nodered there quite easily (already tried to do a NR call on a dummy Lovelace card, and that worked). Are there any downsides to creating the static config? Don't want to spend much time editing them right now, just to realize that I have to regenerate them later for some reason.

**Reply by PiotrMachowski** [reply]
The downside is that when I alter the generator to include more features your instance won't get them out of the box. My advice is to generate static config, but keep only parts that you need to override to make other stuff autogenerated
